src/Bicep.LangServer.IntegrationTests/CompletionTests.cs

```diff
@@ -1470,7 +1471,296 @@ public async Task List_functions_support_completions_for_az_types()
+
+            var bicepFile = SourceFileFactory.CreateBicepFile(new Uri("file:///main.bicep"), file);
+            using var helper = await LanguageServerHelper.StartServerWithTextAsync(TestContext, file, bicepFile.FileUri, creationOptions: new LanguageServer.Server.CreationOptions(NamespaceProvider: NamespaceProvider));
+            var client = helper.Client;
+
+            var completions = await client.RequestCompletion(new CompletionParams
+            {
+                TextDocument = new TextDocumentIdentifier(bicepFile.FileUri),
+                Position = TextCoordinateConverter.GetPosition(bicepFile.LineStarts, cursors[0]),
+            });
+
+            completions.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public async Task VerifyCompletionRequestAfterPoundSign_WithTextBeforePoundSign_ShouldDoNothing()
+        {
+            var fileWithCursors = @"var terminatedWithDirective = 'foo' #|";
+            var (file, cursors) = ParserHelper.GetFileWithCursors(fileWithCursors);
+
+            var bicepFile = SourceFileFactory.CreateBicepFile(new Uri("file:///main.bicep"), file);
+            using var helper = await LanguageServerHelper.StartServerWithTextAsync(TestContext, file, bicepFile.FileUri, creationOptions: new LanguageServer.Server.CreationOptions(NamespaceProvider: NamespaceProvider));
+            var client = helper.Client;
+
+            var completions = await client.RequestCompletion(new CompletionParams
+            {
+                TextDocument = new TextDocumentIdentifier(bicepFile.FileUri),
+                Position = TextCoordinateConverter.GetPosition(bicepFile.LineStarts, cursors[0]),
+            });
+
+            completions.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public async Task VerifyCompletionRequestAfterPoundSign_WithWhiteSpaceBeforePoundSign_ShouldReturnCompletionItem()
+        {
+            var fileWithCursors = @"    #|
+param storageAccount string = 'testAccount";
+            var (file, cursors) = ParserHelper.GetFileWithCursors(fileWithCursors);
+
+            var bicepFile = SourceFileFactory.CreateBicepFile(new Uri("file:///main.bicep"), file);
+            using var helper = await LanguageServerHelper.StartServerWithTextAsync(this.TestContext, file, bicepFile.FileUri, creationOptions: new LanguageServer.Server.CreationOptions(NamespaceProvider: NamespaceProvider));
+            var client = helper.Client;
+
+            var completions = await client.RequestCompletion(new CompletionParams
+            {
+                TextDocument = new TextDocumentIdentifier(bicepFile.FileUri),
+                Position = TextCoordinateConverter.GetPosition(bicepFile.LineStarts, cursors[0]),
+            });
+
+            completions.Should().SatisfyRespectively(
+                c =>
+                {
+                    c.Label.Should().Be(LanguageConstants.DisableNextLineDiagnosticsKeyword);
+                });
```

Comment:
FYI, I recently added a few helpers that simplify some of these tests. You might need to merge main into your branch to pick them up.
* Ability to assert on a single completion (to avoid having to use `SatisfyRespectively`):
 
 
 
 https://github.com/Azure/bicep/blob/8bdc549c3d48573715e6b4295afc5fb2e93f5616/src/Bicep.LangServer.IntegrationTests/CompletionTests.cs#L1272-L1274
* Also note in the above, there's a helper method `RequestCompletion` so you don't have to build the request body yourself
* Ability to assert on what the program text looks like after the completion has been applied (helps visually assert that the completion has the expected impact on the document):
    https://github.com/Azure/bicep/blob/8bdc549c3d48573715e6b4295afc5fb2e93f5616/src/Bicep.LangServer.IntegrationTests/CompletionTests.cs#L1276-L1283

Reply:
Thanks for letting me know about the latest updates. I made the suggested changes here:
https://github.com/Azure/bicep/commit/0baf494238072f53aabbf1cd44a886df84ab0847

Fixed am issue in lexer as well.